src/neo/Wallets/Wallet.cs

```diff
@@ -119,14 +119,24 @@ protected Wallet(string path, ProtocolSettings settings)
         /// <returns>The created account.</returns>
         public WalletAccount CreateAccount()
         {
+        generate:
             byte[] privateKey = new byte[32];
-            using (RandomNumberGenerator rng = RandomNumberGenerator.Create())
+            try
             {
-                rng.GetBytes(privateKey);
+                using (RandomNumberGenerator rng = RandomNumberGenerator.Create())
+                {
+                    rng.GetBytes(privateKey);
```

Comment:
I'm not sure the resulting `privateKey` is checked against curve order anywhere, BTW.

Reply:
If the produced publicKey it's invalid it should be an invalid privateKey. Maybe we can sign and verify a random number in order to ensure that it's a valid privateKey.